### PR TITLE
Remove global var NumeroChroniquesTireesParGroup

### PR DESCRIPTION
**Pull request by payetvin**
#1615

### CODE REVIEW

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 1]
```diff
@@ -319,16 +319,25 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
     const auto constraintCount = activeContraints.size();
     for (unsigned constraintIndex = 0; constraintIndex != constraintCount; ++constraintIndex)
     {
         auto bc = activeContraints[constraintIndex];
         assert(bc->RHSTimeSeries().width && "Invalid constraint data width");
+        uint tmpts = 0;
+        for (const auto& group : bcgroup)
+        {
+            if (group->name() == bc->group())
+                tmpts = group->timeseriesNumbers[0][problem.year];
```
Comment (a-zakir) [inline src/solver/simulation/sim_calcul_economique.cpp]: i think you need a `break` instruction

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 2]
```diff
@@ -319,16 +319,25 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
+                               const uint weekFirstDay, int pasDeTemps)
 {
     auto activeContraints = bindingConstraints.activeContraints();
     const auto constraintCount = activeContraints.size();
     for (unsigned constraintIndex = 0; constraintIndex != constraintCount; ++constraintIndex)
     {
         auto bc = activeContraints[constraintIndex];
         assert(bc->RHSTimeSeries().width && "Invalid constraint data width");
+        uint tmpts = 0;
+        for (const auto& group : bcgroup)
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_calcul_economique.cpp]: You can get the group with:
auto* group = bcgroup[bc->group()]
if (group != nullptr) {
 ...
}

**Review by JasonMarechal25 — CHANGES_REQUESTED** [review]

**Review by a-zakir — APPROVED** [review]

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 3]
```diff
@@ -318,17 +318,29 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
     problem.LeProblemeADejaEteInstancie = false;
 }
 
-void preparerBindingConstraint(const PROBLEME_HEBDO &problem, uint numSpace, int PasDeTempsDebut,
-                               const BindingConstraintsRepository &bindingConstraints, const uint weekFirstDay, int pasDeTemps) 
+void preparerBindingConstraint(const PROBLEME_HEBDO &problem, int PasDeTempsDebut,
+                               const BindingConstraintsRepository &bindingConstraints,
+                               const BindingConstraintGroupRepository &bcgroup,
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_calcul_economique.cpp]: A repository contains several element.
bcgroup would be better name either as a plurial or with "repository" in its name
E.g. bcGroups, bcGroupsRepository, bindingConstrainsGroups, bindingConstraintsGroupsRepository

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 4]
```diff
@@ -318,17 +318,29 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
     auto activeContraints = bindingConstraints.activeContraints();
     const auto constraintCount = activeContraints.size();
     for (unsigned constraintIndex = 0; constraintIndex != constraintCount; ++constraintIndex)
     {
         auto bc = activeContraints[constraintIndex];
         assert(bc->RHSTimeSeries().width && "Invalid constraint data width");
+        uint tmpts = 0;
+        for (const auto& group : bcgroup)
+        {
+            if (group->name() == bc->group())
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_calcul_economique.cpp]: If no groups is found, tmpts = 0
Is it ok?

File: src/solver/simulation/sim_calcul_economique.cpp [hunk 5]
```diff
@@ -318,17 +318,29 @@ void SIM_InitialisationProblemeHebdo(Data::Study& study,
...
+                               const BindingConstraintGroupRepository &bcgroup,
+                               const uint weekFirstDay, int pasDeTemps)
 {
     auto activeContraints = bindingConstraints.activeContraints();
     const auto constraintCount = activeContraints.size();
     for (unsigned constraintIndex = 0; constraintIndex != constraintCount; ++constraintIndex)
     {
         auto bc = activeContraints[constraintIndex];
         assert(bc->RHSTimeSeries().width && "Invalid constraint data width");
+        uint tmpts = 0;
```
Comment (JasonMarechal25) [inline src/solver/simulation/sim_calcul_economique.cpp]: Could you please rename tmpts with a more meaningful name please? Maybe "TSNumberToUse" or something like this depending on what you deem fit

**Review by JasonMarechal25 — DISMISSED** [review]

**Pull request merged**